tools/autograd/gen_variable_type.py

```diff
@@ -449,11 +481,22 @@ def preprocess_nn_functions(declarations):
 
         def actual(arg):
             name = arg['name']
+            # To avoid confusion, we bind the mask in backwards
+            # to the name 'grad_input_mask', which is not the same
+            # as the argument naming convention in NN, which is
+            # 'output_mask'.  When we are autogenerating NN
+            # entries, we want to pass in 'grad_input_mask'
+            # as 'output_mask'; thus the substitution here.
+            if name == 'output_mask':
+                return 'grad_input_mask'
             return name if name != 'inplace' else 'false'
 
         actuals = [actual(arg) for arg in bwd['arguments']]
         formula = '{}({})'.format(bwd_name, ', '.join(actuals))
         formula = formula.replace('grad_output', 'grad')
+        # TODO: Ewwwwwww.  This isn't going to work when there are
```

Comment:
I don't think this comment makes sense. This is only for functions defined by THNN. The formula is created programmatically from the names of the arguments -- it's not written in derivatives.yaml

Reply:
Yeah, you are right!